constructor_functions.js

```diff
@@ -0,0 +1,263 @@
+// # Constructor Exercises
+
+// ## Question 1
+
+// a. Write a constructor function called `Person` that has 3 properties: a first name, a last name and a middle name. Create 2 instances of a `Person`. Print one of their first names.
+function Person (firstName, lastName, middleName){
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.middleName = middleName;
+}
+
+let corey = new Person("corey", "ladovsky", "N/A");
+let kev = new Person("kevin","wong","j");
+console.log(corey.firstName);
+
+
+// b. Write a prototype function in `Person` called `fullName` that will return a formatted string of an instance's full name. Call this method on both the instances you created in part a.
+
+Person.prototype.fullName = function (){    
+    return `${this.firstName} ${this.middleName} ${this.lastName}`;
+}
+console.log(corey.fullName());
+console.log(kev.fullName());
+
+
+
+// ## Question 2
+
+// a. Create a constructor function called `Book` that has properties `title`, `author` and `rating`. Create some instances of `Book`.
+function Book (title, author, rating){
+    Object.assign(this, {title, author, rating});
+    // this.title = title;
+    // this.author = author;
+    // this.rating = rating;
+}
+
+let book1 = new Book("Sapiens: A Brief History of Humankind","Yuval Noah Harari", 9.2);
+let book2 = new Book("Radical Candor: How to Get What You Want by Saying What You Mean","Kim Scott", 8.4);
+
+// b. Add a prototype function to `Book` called `isGood` that returns `true` if its rating is greater than or equal to 7
+
+Book.prototype.isGood = function () {
+    if(this.rating >= 7){
+        return true;
+    } else {
+        return false;
+    }
+}
+console.log(book1.isGood());
+console.log(book2.isGood());
+
+
+
+// ## Question 3
+
+// a. Create a `Dog` constructor function with four properties: `name (string), breed (string), mood (string), and hungry (boolean)`.
+function Dog(name, breed, mood, hungry){
+    Object.assign(this, {name, breed, mood, hungry});
+}
+
+let doge = new Dog("Yuki", "Shiba Inu", "Sleepy", false);
+
+// console.log(doge);
+
+// b. Add a prototype function called `playFetch`. It should set the dog's `hungry` property to `true`, set its mood property to `playful`, and log "Ruff!"
+Dog.prototype.playFetch = function() {
+        this.hungry = true;
+        this.mood = "playful";
+        console.log("Ruff!");
+}
+doge.playFetch();
+console.log(doge);
+
+// c. Add a prototype function called `feed`. If the dog is hungry, it should set `hungry` to `false` and print "Woof!" If the dog is not hungry, it should log "The dog doesn't look hungry"
+Dog.prototype.feed = function () {
+    if (this.hungry === true){
```

Comment:
this.hungry evaluates to a boolean already, therefore this.hungry would suffice on it's own.